hazelcast/src/main/java/com/hazelcast/internal/util/counters/Counter.java

```diff
@@ -38,6 +38,15 @@
      */
     void set(long value);
 
+    /**
+     * Sets the current value of the counter
+     * and returns the old value.
+     *
+     * @param newValue the new value.
+     * @return the old value of the counter.
+     */
+    long getAndSet(long newValue);
```

Comment:
nit: similarly, this could be `drain()` without parameters (the actual `newValue` is always `0` anyway)

Reply:
I thought this method might be useful in the future, we use counters in other places too. So, I tried to be as generic as possible here

Reply:
Sure, no prob, let's get this merged then.